File: Baseball_Game.cpp
#include <bits/stdc++.h>
using namespace std;

int foo(vector<string>& operations)
{
    vector<int> stack;
    for (string i : operations)
    {
        if(i == "D" || i == "C" || i == "+")
        {
            if (i == "+") stack.push_back(stack[stack.size()-2] + stack.back());
            else if (i == "C") stack.pop_back();
            else stack.push_back(stack.back() *2);
        }
        else stack.push_back(stoi(i));
    }
    return accumulate(stack.begin(), stack.end(), 0);
}

File: Baseball_Game.py
operations = ["5","2","C","D","+"]

def foo(operations: list[str]) -> int:
    stack = []
    for i in operations:
        if i not in ["+", "C", "D"]:
            stack.append(int(i))
        elif i == "+":
            stack.append(int(stack[-2]) + int(stack[-1]))
        elif i == "C":
            stack.pop()
        else:
            stack.append(int(stack[-1])*2)
    return sum(stack)

print(foo(operations))